core/src/telemetry/otel.rs

```diff
@@ -359,3 +372,26 @@ fn metric_temporality_to_selector(t: MetricTemporality) -> impl TemporalitySelec
         MetricTemporality::Delta => ConstantTemporality(Temporality::Delta),
     }
 }
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use super::*;
+    use opentelemetry::Key;
+    use std::env;
+
+    #[test]
+    pub(crate) fn default_resource_instance_respects_otel_env_vars() {
```

Comment:
I don't know enough about Rust, do these tests run concurrently? If so, we need to turn it off during this test since this has global side effects (I would also be ok not having this test if others decide that)

Reply:
Yeah I've never used env vars in rust tests before, I'll just remove it as it's basically testing `opentelemetry-sdk` functionality anyway. Was just nice to see it pass as it was the whole point of this PR :)